# Patch to check for GTest and Gflags in a visible way,
# add GTEST_BOTH_LIBRARIES for snappy-unittest; and
# set properly VERSION and SOVERSION properties to produce right symlinks
#
# Copyright (C) 2017 Georgi D. Sotirov <gdsotirov@dir.bg>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License
# as published by the Free Software Foundation; either version 2
# of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write to the Free Software
# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
#
# $Id: snappy-1.1.5.diff,v 1.3 2017/07/06 13:21:42 gsotirov Exp $
#
diff -urNad snappy-1.1.5-orig/CMakeLists.txt snappy-1.1.5/CMakeLists.txt
--- snappy-1.1.5-orig/CMakeLists.txt	2017-06-29 04:37:30.000000000 +0300
+++ snappy-1.1.5/CMakeLists.txt	2017-07-06 16:16:56.267728321 +0300
@@ -56,12 +56,12 @@
 CHECK_CXX_SOURCE_COMPILES("int main(void) { return __builtin_ctzll(0); }"
         HAVE_BUILTIN_CTZ)
 
-FIND_PACKAGE(GTest QUIET)
+FIND_PACKAGE(GTest)
 IF(GTEST_FOUND)
     SET(HAVE_GTEST 1)
 ENDIF()
 
-FIND_PACKAGE(Gflags QUIET)
+FIND_PACKAGE(Gflags)
 IF(GFLAGS_FOUND)
     SET(HAVE_GFLAGS 1)
 ENDIF()
@@ -107,7 +107,7 @@
         ARCHIVE DESTINATION lib)
 INSTALL(EXPORT SnappyTargets NAMESPACE Snappy:: DESTINATION lib/cmake/Snappy)
 
-SET_TARGET_PROPERTIES(snappy PROPERTIES SOVERSION ${PROJECT_VERSION})
+SET_TARGET_PROPERTIES(snappy PROPERTIES VERSION ${PROJECT_VERSION} SOVERSION ${PROJECT_VERSION_MAJOR})
 
 SET(INCLUDE_INSTALL_DIR include)
 SET(LIBRARY_INSTALL_DIR lib)
@@ -146,7 +146,7 @@
 ADD_EXECUTABLE(snappy-unittest snappy_unittest.cc snappy-test.cc)
 TARGET_COMPILE_DEFINITIONS(snappy-unittest PRIVATE -DHAVE_CONFIG_H)
 TARGET_LINK_LIBRARIES(snappy-unittest snappy ${COMPRESSION_LIBS}
-                      ${GFLAGS_LIBRARIES})
+                      ${GFLAGS_LIBRARIES} ${GTEST_BOTH_LIBRARIES})
 TARGET_INCLUDE_DIRECTORIES(snappy-unittest BEFORE PRIVATE ${Snappy_SOURCE_DIR}
                            ${GTEST_INCLUDE_DIRS} ${GFLAGS_INCLUDE_DIRS})
 
